deploy-service/common/src/main/java/com/pinterest/deployservice/rodimus/RodimusManagerImpl.java

```diff
@@ -68,31 +73,98 @@ public boolean shouldSkipClass(Class<?> c) {
+    }
+
     @Override
     public void terminateHostsByClusterName(String clusterName, Collection<String> hostIds) throws Exception {
         if (hostIds.isEmpty()) {
             return;
         }
         
-        String url = String.format("%s/v1/clusters/%s/hosts", rodimusUrl, clusterName);
-        httpClient.delete(url, gson.toJson(hostIds), headers, RETRIES);
-    }
+        int rtry = this.RETRIES;
+        do{
+//            LOG.info("terminateHostsByClusterName");
+            String url = String.format("%s/v1/clusters/%s/hosts", this.rodimusUrl, clusterName);
+            setAuthorization();
+            try {
+                httpClient.delete(url, gson.toJson(hostIds), this.headers, RETRIES);
+                return;
+            } catch (DeployInternalException e) {
+                if ( ! this.refreshCachedKey() ) throw e;
+                rtry--;
+            }
+        }while( rtry>0 );
+        throw new DeployInternalException(
+            "HTTP request to Rodimus API failed on terminateHostsByClusterName, too many retries.");
+    } // terminateHostsByClusterName
 
     @Override
     public Collection<String> getTerminatedHosts(Collection<String> hostIds) throws Exception {
         if (hostIds.isEmpty()) {
             return Collections.emptyList();
         }
-        // NOTE: it's better to call this function with single host id
-        String url = String.format("%s/v1/hosts/state?actionType=%s", rodimusUrl, "TERMINATED");
-        String res = httpClient.post(url, gson.toJson(hostIds), headers, RETRIES);
-        return gson.fromJson(res, new TypeToken<ArrayList<String>>() {}.getType());
-    }
+
+        int rtry = this.RETRIES;
+        do{
+//            LOG.info("getTerminatedHosts");
+            // NOTE: it's better to call this function with single host id
+            String url = String.format("%s/v1/hosts/state?actionType=%s", rodimusUrl, "TERMINATED");
+            setAuthorization();
+            try {
+                String res = httpClient.post(url, gson.toJson(hostIds), this.headers, this.RETRIES);
+                return gson.fromJson(res, new TypeToken<ArrayList<String>>() {}.getType());
+            } catch (DeployInternalException e) {
+                if ( ! this.refreshCachedKey() ) throw e;
+                rtry--;
+            }
+        }while( rtry>0 );
+        throw new DeployInternalException(
+            "HTTP request to Rodimus API failed on getTerminatedHosts, too many retries.");
+    } // getTerminatedHosts
 
     @Override
     public Long getClusterInstanceLaunchGracePeriod(String clusterName) throws Exception {
-        String url = String.format("%s/v1/groups/%s/config", rodimusUrl, clusterName);
-        String res = httpClient.get(url, null, null, headers, RETRIES);
+        String res = null;
+
+        int rtry = this.RETRIES;
+        do{
+//            LOG.info("getClusterInstanceLaunchGracePeriod");
```

Comment:
why logs are commented out. Either log proper messages or if logging is not required then remove dead code.